zerver/lib/avatar.py

```diff
@@ -36,6 +34,33 @@ def avatar_url(
     )
 
 
+def get_system_bots_avatar_file_name(email: str) -> str:
+    # System bot avatar files are stored in the `images/static_avatars/`
+    # directory as .png files. We can determine which avatar belongs
+    # to which bot by matching the bot's email username with the avatar
+    # file name.
+    # e.g., notification-bot@zulip.com -> notification-bot.png
+    return STATIC_AVATARS_DIR + email.split("@")[0]
+
+
+def get_static_avatar_url(email: str, medium: bool) -> str:
+    avatar_file_name = get_system_bots_avatar_file_name(email)
+
+    avatar_file_name += "-medium.png" if medium else ".png"
+
+    if not DEBUG and not staticfiles_storage.exists(avatar_file_name):
+        # This logic cannot be executed while STATIC_ROOT is
+        # not defined.
+        return DEFAULT_AVATAR_FILE
+    else:
+        from django.contrib.staticfiles.finders import find
+
+        if not find(avatar_file_name):
+            raise AssertionError(f"Unknown avatar file: {avatar_file_name}")
```

Comment:
This is just a net to catch problems early on if any breaking changes to this system. For example, the avatar file name got changed, they moved, the system bots user email got updated, etc.